psi4/src/psi4/libscf_solver/rhf.cc

```diff
@@ -937,5 +937,34 @@ bool RHF::stability_analysis() {
     // FOLLOW is not implemented for RHF
     return false;
 }
+
+std::shared_ptr<RHF> RHF::c1_deep_copy(std::shared_ptr<BasisSet> basis)
+{
+    std::shared_ptr<Wavefunction> wfn = Wavefunction::c1_deep_copy(basis);
+    auto hf_wfn = std::make_shared<RHF>(wfn, functional_, wfn->options(), wfn->psio());
+    // now just have to copy the matrices that RHF initializes
+    // include only those that are not temporary (some deleted in finalize())
+    if (Ca_) hf_wfn->Ca_ = Ca_subset("AO", "ALL");
+    hf_wfn->Cb_ = hf_wfn->Ca_;
+    if (Da_) hf_wfn->Da_ = Da_subset("AO");
+    hf_wfn->Db_ = hf_wfn->Da_;
+    hf_wfn->D_  = hf_wfn->Da_;
```

Comment:
Won't `hf_wfn->Da_` not be available if `!Da_`. Maybe the logic needs reoorganization, even though the absence of `Da_` is unlikely.

Reply:
Good catch, fixing that